File: ceylon-travels/src/services/CustomerService.js
import axios from 'axios';

const Customer_API_Base_Url = "http://localhost:8087/customers";

class CustomerService{
    getCustomers(){
        return axios.get(Customer_API_Base_Url);
    }

    createCustomer(customer){
        return axios.post(Customer_API_Base_Url, customer);
    }

    getCustomerById(id){
        return axios.get(Customer_API_Base_Url+ '/' +id);
    }

    updateCustomer(customer){
        return axios.put(Customer_API_Base_Url, customer);
    }

    deleteCustomer(customerid){
        return axios.delete(Customer_API_Base_Url + '/' + customerid);
    }
}

export default new CustomerService()

File: Rest APIs/customers/customer/src/main/java/com/example/customer/service/customerService.java
package com.example.customer.service;

import com.example.customer.data.customer;
import com.example.customer.data.customerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class customerService {

    //Using Autowired to Automatically inject the necessary obj
    @Autowired
    private customerRepository cusRepo;


    public List<customer> getAllCustomer(){
        List<customer> customers = cusRepo.findAll();
        return customers;
    }

    public customer getCustomerById(int id){
        Optional<customer> custm = cusRepo.findById(id);
        return custm.get();
    }
    public customer createCustomer(customer cus){
        return cusRepo.save(cus);
    }

    public customer updateCustomer(customer cust_up){
        if(getCustomerById(cust_up.getId())!=null) {
            return cusRepo.save(cust_up);
        }
        else {
            return cusRepo.save(cust_up);
        }
    }

    public void deleteCustomer(int custm){
        if(getCustomerById(custm)!=null){
            cusRepo.deleteById(custm);
        }
    }

}
